machine.go

```diff
@@ -616,8 +616,17 @@ func (m *Machine) captureFifoToFile(logger *log.Entry, fifoPath string, w io.Wri
 			}
 		}()
 
-		if _, err := io.Copy(w, fifoPipe); err != nil {
-			logger.Warnf("io.Copy failed to copy contents of fifo pipe: %v", err)
+		var err error
+		// since we are opening the fifo with O_NONBLOCK, any reads on a not
+		// connected fifo will return immediately with no error. To prevent nothing
+		// being read to the io.Writer, we will retry until an error has occurred
+		// which happens when the application has exited or the context has been
+		// closed.
+		for err == nil {
+			if _, err = io.Copy(w, fifoPipe); err != nil {
+				logger.Warnf("io.Copy failed to copy contents of fifo pipe: %v", err)
```

Comment:
If we're retrying past EOF and we expect an error, we're going to start getting a warning every time.  Maybe we don't want to log a warning when we're expecting an error?

Reply:
@samuelkarp - I reworked this to use `syscall.Select`. This is a much better solution. Let me know what you think!